int32 handle
---
bool hasSucceeded